test/test_comment.py

```diff
@@ -0,0 +1,197 @@
+# limitations under the License.
+
+"""Test the keyword argument 'comment' in various helpers."""
+
+import sys
+import inspect
+from typing import Any, Union
+from collections import defaultdict
+
+sys.path[0:0] = [""]
+
+from pymongo.mongo_client import MongoClient
+from pymongo.database import Database
+from pymongo.collection import Collection
+from pymongo.read_concern import ReadConcern
+from pymongo.read_preferences import ReadPreference
+from pymongo.write_concern import WriteConcern
+from pymongo.command_cursor import CommandCursor
+from pymongo.operations import IndexModel
+from bson.dbref import DBRef
+
+from test import client_context, SkipTest, unittest, IntegrationTest
+from test.utils import rs_or_single_client, EventListener
+
+
+class Empty(object):
+    def __getattr__(self, item):
+        try:
+            self.__dict__[item]
+        except KeyError:
+            return self.empty
+
+    def empty(self, *args, **kwargs):
+        return Empty()
+
+
+class TestComment(IntegrationTest):
+    def _test_ops(self, helpers, already_supported, listener, db=Empty(), coll=Empty()):
+        results = listener.results
+        for h, args in helpers:
+            c = "testing comment with " + h.__name__
+            with self.subTest("collection-" + h.__name__ + "-comment"):
+                for cc in [c, {"key": c}, ["any", 1]]:
+                    results.clear()
+                    kwargs = {"comment": cc}
+                    if h == coll.rename:
+                        tmp = db.get_collection("temp_temp_temp").drop()
+                        destruct_coll = db.get_collection("test_temp")
+                        destruct_coll.insert_one({})
+                        maybe_cursor = destruct_coll.rename(*args, **kwargs)
+                        destruct_coll.drop()
+                    elif h == db.validate_collection:
+                        coll = db.get_collection("test")
+                        coll.insert_one({})
+                        maybe_cursor = db.validate_collection(*args, **kwargs)
+                    else:
+                        coll.create_index("a")
+                        maybe_cursor = h(*args, **kwargs)
+                    self.assertIn(
+                        "comment",
+                        inspect.signature(h).parameters,
+                        msg="Could not find 'comment' in the "
+                        "signature of function %s" % (h.__name__),
+                    )
+                    self.assertEqual(
+                        inspect.signature(h).parameters["comment"].annotation, Union[Any, None]
+                    )
+                    if isinstance(maybe_cursor, CommandCursor):
+                        maybe_cursor.close()
+                    tested = False
+                    for i in results["started"]:
+                        if cc == i.command.get("comment", ""):
+                            tested = True
```

Comment:
Can we assert the comment appears on all the commands?:
```
                    for event in results["started"]:
                        cmd = event.command
                        self.assertIn("comment", cmd)
                        self.assertEqual(cc, cmd["comment"])
```

Reply:
Done.

Reply:
This for some reason is not possible with `list_indexes`. I will investigate why.

Reply:
Fixed the underlying issue.